packages/solid-notifications/services/provider/sub-services/notification-channel.mixin.js

```diff
@@ -0,0 +1,313 @@
+      const { containerUri, resourceUri } = ctx.params;
+      this.onContainerOrCollectionEvent(containerUri, resourceUri, ACTIVITY_TYPES.REMOVE);
+    },
+    async 'activitypub.collection.added'(ctx) {
+      const { collectionUri, itemUri } = ctx.params;
+      this.onContainerOrCollectionEvent(collectionUri, itemUri, ACTIVITY_TYPES.ADD);
+    },
+    async 'activitypub.collection.removed'(ctx) {
+      const { collectionUri, itemUri } = ctx.params;
+      this.onContainerOrCollectionEvent(collectionUri, itemUri, ACTIVITY_TYPES.REMOVE);
+    }
+  },
+  methods: {
+    async getModified(resourceUri) {
+      return await this.broker.call('ldp.resource.get', { resourceUri, webId: 'system' })?.['dc:modified'];
+    },
+    getMatchingChannels(topic) {
+      const now = new Date();
+      const matchedChannels = this.channels
+        .filter(c => c.topic === topic)
+        .filter(c => (c.startAt ? new Date(c.startAt) <= now : true))
+        .filter(c => (c.endAt ? new Date(c.endAt) > now : true))
+        // Check if rate is exceeded.
+        .filter(c => {
+          if (!(c.lastTriggered && c.rate)) return true;
+          return moment.duration(c.rate).asMilliseconds() < now - c.lastTriggered;
+        });
+
+      return matchedChannels;
+    },
+    onContainerOrCollectionEvent(containerOrCollectionUri, resourceOrItemUri, type) {
+      const activity = {
+        '@context': ['https://www.w3.org/ns/activitystreams', 'https://www.w3.org/ns/solid/notifications-context/v1'],
+        id: `urn:uuid:${uuidV4()}`,
+        type,
+        object: resourceOrItemUri,
+        target: containerOrCollectionUri
+      };
+      this.triggerChannelsForTopic(containerOrCollectionUri, activity);
+    },
+    onResourceEvent(resourceUri, type, state) {
+      const activity = {
+        '@context': ['https://www.w3.org/ns/activitystreams', 'https://www.w3.org/ns/solid/notifications-context/v1'],
+        id: `urn:uuid:${uuidV4()}`,
+        type,
+        object: resourceUri,
+        state
+      };
+      this.triggerChannelsForTopic(resourceUri, activity);
+    },
+    async triggerChannelsForTopic(topicUri, activity) {
+      const channelsToTrigger = this.getMatchingChannels(topicUri);
+
+      // First set lastTriggered, so channels with rate are not triggered multiple times.
+      const now = new Date();
+      for (const channel of channelsToTrigger) {
+        channel.lastTriggered = now;
+      }
+
+      // Trigger onEvent for each channel (handled by implementing service).
+      await Promise.all(
+        channelsToTrigger.map(async channel => {
+          await this.onEvent(channel, activity);
+        })
+      );
+    },
+    async loadChannelsFromDb({ removeOldChannels }) {
+      // Load all channels from all Pods
+      await Promise.all(
+        (await this.broker.call('pod.list')).map(async dataset => {
+          // Why not use auth.account.find here?
+          const webId = urlJoin(this.settings.baseUrl, dataset);
```

Comment:
Indeed. These two actions are some kind of duplicates. The `auth.account.find` returns more informations but is maybe slightly slower. We should probably stick to `auth.account.find` since it also works on non-Pod environment.